src/Microsoft.DotNet.Darc/Darc/Options/VirtualMonoRepo/VmrPushCommandLineOptions.cs

```diff
@@ -22,5 +22,5 @@ internal class VmrPushCommandLineOptions : VmrCommandLineOptions
     [Option("commit-verification-pat", Required = false, HelpText = "Token for authenticating to GitHub GraphQL API. Needs to have only basic scope as it will be used to look for commits in public GitHub repos.")]
     public string CommitVerificationPat { get; set; }
 
-    public override Operation GetOperation() => new PushOperation(this);
+    public override Type GetOperation() => typeof(PushOperation);
```

Comment:
Why we can't keep this returning an Operation which it would resolve from the DI?

It would also mean we don't need to cast the options objects all the time.

It would be fairly easy, we would use `ActivatorUtilities` to create the operation and pass `this` as the additional arguments. The operations could then have the right option type in the ctor.

Reply:
Yeah this is probably a simpler approach